src/proto/FunctionRpc.proto

```diff
@@ -548,11 +548,21 @@ message RpcException {
   // Source of the exception
   string source = 3;
 
-  // Stack trace for the exception
+  // Stack trace for the exception. If it's a user code exception,
+  // worker should assign this value to the that exception's stack trace, 
+  // so that it can be passed along to app insights.  
   string stack_trace = 1;
 
   // Textual message describing the exception
   string message = 2;
+
+  // Worker specifies whether exception is a user exception, 
+  // for purpose of application insights logging. Defaults to false. 
+  optional bool is_user_exception = 4;
```

Comment:
You're the first person to use `optional` in this file. I assume that's fine? But just surprised it hasn't been used already

Reply:
Should be okay, we didn't want to have a breaking change xD

Reply:
Yes, maybe because these are the only fields on a message definition that are used in one specific scenario and not in others- we don't want to require that workers explicitly set these fields for exceptions not thrown by user code.

Reply:
Ah if they're optional by default we should remove the label